dagshub/models/model_locator.py

```diff
@@ -0,0 +1,320 @@
+                bucketPath = f"{storage.protocol}/{str_path}"
+                return bucketPath, StorageType.Bucket
+        return path, StorageType.Repo
+
+    def try_load_yaml(self, yaml_path=".dagshub/model.yaml") -> Optional[ModelLoader]:
+        """
+        Loads the info about where the model stored from a .yaml file
+        """
+        try:
+            yaml_contents = self.repo_api.get_file(yaml_path)
+        except PathNotFoundError:
+            return None
+        cfg = dacite.from_dict(ModelYamlStruct, yaml.safe_load(yaml_contents))
+
+        model_path, storage_type = self._handle_path(cfg.model_dir)
+        log_message(f"Loading the model from yaml file {yaml_path}", logger=logger)
+        if storage_type == StorageType.DagshubStorage:
+            try:
+                self.repo_api.list_storage_path(f"s3/{self.repo_api.repo_name}/{model_path}")
+                log_message(f"Loading the model from DagsHub Storage {model_path}", logger=logger)
+                return DagsHubStorageModelLoader(self.repo_api, PurePosixPath(model_path))
+            except PathNotFoundError:
+                raise ModelNotFoundError
+        elif storage_type == StorageType.Bucket:
+            try:
+                self.repo_api.list_storage_path(model_path)
+                log_message(f"Loading the model from bucket {model_path}", logger=logger)
+                return BucketModelLoader(self.repo_api, PurePosixPath(model_path))
+            except PathNotFoundError:
+                raise ModelNotFoundError
+        else:
+            try:
+                self.repo_api.list_path(model_path, revision=self.git_ref)
+                log_message(f"Loading the model from repo {model_path}", logger=logger)
+                return RepoModelLoader(self.repo_api, self.git_ref, PurePosixPath(model_path))
+            except PathNotFoundError:
+                raise ModelNotFoundError
+
+    lookup_dirs = ["model", "models"]
+
+    def try_find_model_in_repo(self) -> Optional[ModelLoader]:
+        for lookup_dir in self.lookup_dirs:
+            modelPath = PurePosixPath(lookup_dir)
+            try:
+                self.repo_api.list_path(str(modelPath), revision=self.git_ref)
+                log_message(f"Loading the model from repo directory {modelPath}", logger=logger)
+                return RepoModelLoader(repo_api=self.repo_api, revision=self.git_ref, path=modelPath)
+            except PathNotFoundError:
+                continue
+        return None
+
+    def try_find_model_in_bucket(self, bucket: str) -> Optional[ModelLoader]:
+        bucketPath: Optional[PurePosixPath] = None
+        for storage in self.repo_storages:
+            if storage.name.startswith(bucket):
+                bucketPath = PurePosixPath(f"{storage.protocol}/{storage.name}")
+                break
+
+        if bucketPath is None:
+            return None
+
+        for lookup_dir in self.lookup_dirs:
+            modelPath = bucketPath / lookup_dir
```

Comment:
The model path always comes as a directory?

Reply:
Yes, it's intended to always be a directory.
We obviously can't stop users at pointing at files, but the way HuggingFace works is that you point at dirs